metagraph/src/graph/representation/base/dbg_wrapper.hpp

```diff
@@ -148,6 +154,9 @@ class DBGWrapper : public DeBruijnGraph {
   protected:
     std::shared_ptr<const Graph> graph_;
     std::shared_ptr<Graph> graph_ptr_;
+
+    // clear any internal storage the wrapper may have
+    virtual void flush() {}
```

Comment:
This is too much. Better overload method `load` in the children.

Reply:
this method is called by `add_sequence` and `load`, so I designed it this way so that the children would not have to overload them

Reply:
I see, but this looks bad. You make it less more confusing by adding any methods which aren't in `DeBruijnGraph`.

The abstract wrapper should not have any new methods.

Reply:
You are essentially filling holes with hacks like this.

But you will never fill all possible holes. So if you stick to the same strategy, this class will grow and grow in the future, which is obviously not good.

Reply:
I figured that this is something that would be needed by any wrapper which stores auxiliary information, so I don't really see it as a hack.

One pitfall, though, is that I think it should be a pure abstract function in `DBGWrapper` since it needs to be overridden in most cases anyway. If it's like this, then it'll be easy to forget to do this.

Reply:
Hm, I think wrappers with aux stuff should override loaders/etc

Reply:
anyway, I don't think it's worth arguing over this too much...

I removed `flush`. Instead, I've made it so that `add_sequence` and `load` are pure virtual in `DBGWrapper` to remind the programmer to update their internal storage whenever the underlying graph is modified.